query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusRestClient.java

```diff
@@ -14,15 +14,22 @@
 import okhttp3.Response;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 
-public class PrometheusRestClient {
+class PrometheusRestClient {
   private static final String INSTANT_QUERY = "api/v1/query";
   private static final String RANGE_QUERY = "api/v1/query_range";
 
-  private String host;
-  private int port;
-  private OkHttpClient okHttpClient;
+  private final String host;
+  private final int port;
+  private final OkHttpClient okHttpClient;
 
-  public PrometheusRestClient(String host, int port) {
+  PrometheusRestClient(String connection) {
```

Comment:
Why take a connection string that we need to understand the format of? Seemed better before

Reply:
This is driven from the config itself, where we are taking a string called `connectionString`, so just wanted to stick with that in code as well.

Reply:
Isn't that our config or am I misunderstanding? Why not just take them as a distinct host and port like we do everywhere else?

Reply:
> Why not just take them as a distinct host and port like we do everywhere else

We don't do it that way in here. 

When the app config is read, the client config is represented by `RequestHandlerClientConfig` inner class from https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryServiceConfig.java#L78, 

```
  public static class RequestHandlerClientConfig {
    private static final String CONFIG_PATH_TYPE = "type";
    private static final String CONFIG_PATH_CONNECTION_STRING = "connectionString";
    private String type;
    private String connectionString;
```

Now I could have modified it to add host and port additionally and only use them for prom, but the current approach felt better to me. What do you say?

Reply:
OK got it - the pinot version uses a connection string (because it uses zookeeper and is not just a host/port pair), and the way the handler config is architected, all handlers regardless of type share config classes. I think the shared code between different handler types is problematic (for reasons like this), but that's a much broader problem and far out of scope.

Resolving.